### PR TITLE
feat: 🎸 update the modification date of root dataset dir

### PR DESCRIPTION
**Pull request by severo**
in cached assets. It will help delete old directories.

### CODE REVIEW

File: libs/libcommon/src/libcommon/viewer_utils/asset.py [hunk 1]
```diff
@@ -42,6 +42,9 @@ def update_last_modified_date_of_rows_in_assets_dir(
     length: int,
     assets_directory: StrPath,
 ) -> None:
+    namespace, repo = dataset.split('/')
+    root_dir = namespace if repo else dataset
+    (Path(assets_directory).resolve() / root_dir).touch()
```
Comment (lhoestq) [inline libs/libcommon/src/libcommon/viewer_utils/asset.py:47]: this creates a file, it doesn't update the modified time of the directory afaik

Comment (severo) [reply libs/libcommon/src/libcommon/viewer_utils/asset.py:47]: oh, ok. let me fix that

Comment (lhoestq) [reply libs/libcommon/src/libcommon/viewer_utils/asset.py:47]: you can try this

```python

def update_last_modified_date_of_rows_in_assets_dir(
    dataset: str,
    config: str,
    split: str,
    offset: int,
    length: int,
    assets_directory: StrPath,
) -> None:
    dataset_path = Path(assets_directory).resolve() / dataset
    if dataset_path.is_dir():
        # update the directory's last modified date
        if (dataset_path / DATASETS_SERVER_MDATE_FILENAME).is_file():
            try:
                (dataset_path / DATASETS_SERVER_MDATE_FILENAME).unlink()
            except FileNotFoundError:
                pass
    if "/" in dataset:
        namespace_path = Path(assets_directory).resolve() / dataset.split("/")[0]
        if namespace_path.is_dir():
            # update the directory's last modified date
            if (namespace_path / DATASETS_SERVER_MDATE_FILENAME).is_file():
                try:
                    (namespace_path / DATASETS_SERVER_MDATE_FILENAME).unlink()
                except FileNotFoundError:
                    pass
    row_dirs_path = Path(assets_directory).resolve() / dataset / DATASET_SEPARATOR / config / split
    for row_idx in range(offset, offset + length):
        if (row_dirs_path / str(row_idx)).is_dir():
            # update the directory's last modified date
            if (row_dirs_path / str(row_idx) / DATASETS_SERVER_MDATE_FILENAME).is_file():
                try:
                    (row_dirs_path / str(row_idx) / DATASETS_SERVER_MDATE_FILENAME).unlink()
                except FileNotFoundError:
                    pass
            (row_dirs_path / str(row_idx) / DATASETS_SERVER_MDATE_FILENAME).touch()
```

Comment (severo) [reply libs/libcommon/src/libcommon/viewer_utils/asset.py]: fixed

Comment (severo) [reply libs/libcommon/src/libcommon/viewer_utils/asset.py]: Luckily I had not to update the tests because there are none 😛

Comment (severo) [reply libs/libcommon/src/libcommon/viewer_utils/asset.py]: ups: there are tests in services/rows

Comment (severo) [reply libs/libcommon/src/libcommon/viewer_utils/asset.py]: https://github.com/huggingface/datasets-server/actions/runs/5671267524/job/15368026024?pr=1569

File: libs/libcommon/src/libcommon/viewer_utils/asset.py [hunk 2]
```diff
@@ -42,16 +55,12 @@ def update_last_modified_date_of_rows_in_assets_dir(
     length: int,
     assets_directory: StrPath,
 ) -> None:
+    namespace, repo = dataset.split("/")
+    root_dir = namespace if repo else dataset
+    update_directory_modification_date(Path(assets_directory).resolve() / root_dir)
```
Comment (lhoestq) [inline libs/libcommon/src/libcommon/viewer_utils/asset.py:60]: this hsould do the job
```suggestion
    update_directory_modification_date(Path(assets_directory).resolve() / dataset.split("/")[0)
```

File: libs/libcommon/src/libcommon/viewer_utils/asset.py [hunk 3]
```diff
@@ -42,16 +55,12 @@ def update_last_modified_date_of_rows_in_assets_dir(
     length: int,
     assets_directory: StrPath,
 ) -> None:
+    namespace, repo = dataset.split("/")
```
[the remark this answers is not in the document]
Comment (severo) [reply libs/libcommon/src/libcommon/viewer_utils/asset.py:58]: 🤦

**Pull request merged**